### PR TITLE
Fixed null access

### PR DESCRIPTION
**Pull request by zii-dmg**
`parentContext` was checked for null after access to `parentContext.Handler` and `.ReleasePolicy`

### CODE REVIEW

**Comment by jonorossi** [comment]
We really need a unit test for a case where the parent context is null. If this is a bug, it hasn't surfaced in nearly 20 years.

**Comment by zii-dmg** [comment]
Found by static analyzer.

**Comment by zii-dmg** [comment]
It's public API. Anyone can pass null and will get `NullReferenceException` instead of `ArgumentNullException`.

File: src/Castle.Windsor/MicroKernel/Context/CreationContext.cs [hunk 1]
```diff
@@ -67,7 +67,7 @@ public class CreationContext :
 		/// <param name = "parentContext"> The parent context. </param>
 		/// <param name = "propagateInlineDependencies"> When set to <c>true</c> will clone <paramref name = "parentContext" /> <see cref = "AdditionalArguments" /> . </param>
 		public CreationContext(Type requestedType, CreationContext parentContext, bool propagateInlineDependencies)
-			: this(parentContext.Handler, parentContext.ReleasePolicy, requestedType, null, null, parentContext)
+			: this(parentContext?.Handler, parentContext?.ReleasePolicy, requestedType, null, null, parentContext)
```
Comment (jonorossi) [inline src/Castle.Windsor/MicroKernel/Context/CreationContext.cs:70]: The problem with doing this is that you call the other constructor passing a null `IHandler` and `IReleasePolicy` which isn't valid.

You'll find that Windsor likely has a bunch of these "bugs" that aren't hurting anyone because Windsor's original design was to be completely open with everything marked public, you can see `ResolutionContext` in the same file is public with no checking, but is never created by users. We've been slowly closing things with Castle DynamicProxy because we needed to to keep the public API sane and make changes, however Windsor is still completely open.

I'd like to see Windsor enable nullable reference types, however accepting this pull request would mean that the other constructor needs to accept null and run to completion to then run this constructor which throws.

**Review by jonorossi — CHANGES_REQUESTED** [review]
